### ISSUE TITLE
bug(StorageManager/a11y): Upload status is not announced to screen readers

### ISSUE THREAD

**Issue by hbuchel** [aws-amplify/amplify-ui#5317]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Storage (Storage Manager)

### How is your app built?

Next.js

### Please describe your bug.

The change in upload status when uploading a file is not announced to screen readers. I'd expect to hear "Upload in progress", "Uploaded", "Failed upload", etc. when I file's upload status changes.

**Some considerations:**
- Probably use `aria-live="polite"` since there can be multiple file uploads happening at a time.
- It may be better to also include the file name, like "Uploaded file-name.png" in the text that is announced, since there can be multiple files uploaded at a time.

### What's the expected behaviour?

I can hear the file upload status change when it happens via a screen reader.

### Help us reproduce the bug!

Using a screen reader, upload a file using StorageManager and note that there is no status update announced. You likely just hear the upload button announced since it receives focus after the OS dialog closes after choosing a file.

### Code Snippet

```typescript
// Put your code below this line.

```
